src/include/catalog/catalog_entry/index_catalog_entry.h

```diff
@@ -12,38 +13,42 @@ class KUZU_API IndexCatalogEntry : public CatalogEntry {
     //===--------------------------------------------------------------------===//
     IndexCatalogEntry() = default;
 
-    IndexCatalogEntry(common::table_id_t tableID, std::string indexName)
+    IndexCatalogEntry(std::string type, common::table_id_t tableID, std::string indexName)
         : CatalogEntry{CatalogEntryType::INDEX_ENTRY,
               common::stringFormat("{}_{}", tableID, indexName)},
-          tableID{tableID}, indexName{std::move(indexName)} {}
+          type{std::move(type)}, tableID{tableID}, indexName{std::move(indexName)} {}
+
+    std::string getIndexType() const { return type; }
 
     common::table_id_t getTableID() const { return tableID; }
 
+    std::string getIndexName() const { return indexName; }
+
+    uint8_t* getAuxBuffer() const { return auxBuffer.get(); }
+
+    uint64_t getAuxBufferSize() const { return auxBufferSize; }
+
     //===--------------------------------------------------------------------===//
     // serialization & deserialization
     //===--------------------------------------------------------------------===//
```

Comment:
Can you put a comment here to describe how the serialize and deserialize works for indexes as extensions?